### PR TITLE
[SYCL] Test for program::link() function

### PR DESCRIPTION
**Pull request by maximdimakov**
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

File: SYCL/Config/program_link.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  if (p.is_host()) {
+    return 0;
+  }
+  context c(p);
+  program prog1(c);
+  prog1.compile_with_kernel_type<DUMMY>();
+  prog1.link("-cl-finite-math-only");
+  assert(prog1.get_state() == cl::sycl::program_state::linked &&
+         "fail to link program");
+  // CHECK-IS-FINITE-MATH: -cl-fast-relaxed-math
```
Comment (vladimirlaz) [inline SYCL/Config/program_link.cpp]: Check name does not correlate with the option.

File: SYCL/Config/program_link.cpp [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  // CHECK-IS-FINITE-MATH-NOT: -cl-finite-math-only
+  assert(prog1.get_link_options() == "-cl-finite-math-only" &&
+         "program::get_link_options() output is wrong");
+
+  program prog2(c);
+  prog2.compile_with_kernel_type<DUMMY>("-cl-mad-enable");
+  assert(prog2.get_state() == cl::sycl::program_state::compiled &&
+         "fail to compile program");
+  // CHECK-IS-OPT-DISABLE: -cl-opt-disable
+  // CHECK-IS-OPT-DISABLE-NOT: -cl-mad-enable
```
Comment (vladimirlaz) [inline SYCL/Config/program_link.cpp:40]: This check is not used

File: SYCL/Config/program_link.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    return 0;
+  }
+  context c(p);
+  program prog1(c);
+  prog1.compile_with_kernel_type<DUMMY>();
+  prog1.link("-cl-finite-math-only");
+  assert(prog1.get_state() == cl::sycl::program_state::linked &&
+         "fail to link program");
+  // CHECK-IS-FINITE-MATH: -cl-fast-relaxed-math
+  // CHECK-IS-FINITE-MATH-NOT: -cl-finite-math-only
```
Comment (vladimirlaz) [inline SYCL/Config/program_link.cpp]: this check is not used

File: SYCL/Config/program_link.cpp (hunk 2 above)
Comment (maximdimakov) [reply SYCL/Config/program_link.cpp:40]: Directive with "NOT" suffix is automatically checked when it is checked without suffix

File: SYCL/Config/program_link.cpp (hunk 1 above)
Comment (maximdimakov) [reply SYCL/Config/program_link.cpp]: It fixed with the last commit

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**